packages/hawtio/src/plugins/shared/connect-service.ts

```diff
@@ -244,8 +260,15 @@ class ConnectService implements IConnectService {
   redirect() {
     const url = new URL(window.location.href)
     const redirect = url.searchParams.get(PARAM_KEY_REDIRECT) ?? hawtio.getBasePath() ?? '/'
-    log.debug('Redirect to:', redirect)
-    window.location.href = redirect
+    const { hostname, port, protocol } = new URL(redirect)
+
+    //check if it's correct redirect:
+    if (hostname === url.hostname && port === url.port && ['http:', 'https:'].includes(protocol)) {
+      log.debug('Redirect to:', redirect)
+      window.location.href = encodeURI(redirect)
+    } else {
+      log.error('Redirect param modified')
```

Comment:
writing error log is great, but maybe then redirect to `/` (default) or showing the error message to the user with notification?